misaligned.js

```diff
@@ -1,16 +1,45 @@
-const {expect} = require('chai')
+const { expect } = require("chai");
 
-function print_color_map() {
-    const majorColors = ["White", "Red", "Black", "Yellow", "Violet"];
-    const minorColors = ["Blue", "Orange", "Green", "Brown", "Slate"];
-    for (let i = 0; i < majorColors.length; i++) {
-        for (let j = 0; j < minorColors.length; j++) {
-            console.log(`${i * 5 + j} | ${majorColors[i]} | ${minorColors[j]}`);
-        }
+function ColorMap(colorCode, majorColor, minorColor) {
+  this.colorCode = colorCode;
+  this.majorColor = majorColor;
+  this.minorColor = minorColor;
+}
+
+function get_color_map() {
+  const majorColors = ["White", "Red", "Black", "Yellow", "Violet"];
+  const minorColors = ["Blue", "Orange", "Green", "Brown", "Slate"];
+  let colorMaps = [];
+  for (let i = 0; i < majorColors.length; i++) {
+    for (let j = 0; j < minorColors.length; j++) {
+      colorMaps.push(new ColorMap(i * 5 + j + 1, majorColors[i], minorColors[j]));
     }
-    return majorColors.length * minorColors.length;
+  }
+  return colorMaps;
+}
+
+function format_color_map(colorMap) {
+  return `${colorMap.colorCode.toString().padEnd(2,' ')} | ${colorMap.majorColor.padEnd(6,' ')} | ${colorMap.minorColor}`;
 }
 
-result = print_color_map();
-expect(result).equals(25);
-console.log('All is well (maybe!)');
+function print_color_map(colorMaps) {
+  colorMaps.forEach((colorMap) => {
+    console.log(format_color_map(colorMap));
+  });
+}
+
+const result = get_color_map();
+print_color_map(result);
+
+expect(result.length).equals(25, "Color map length should be 25");
+expect(result[0].colorCode).equals(1, "Color code should start from 1");
+expect(format_color_map(result[0])).equals(
+  "1  | White  | Blue",
+  "First color map should be 1 | White | Blue"
+);
+expect(format_color_map(result[5])).equals(

```

Comment:
These tests are checking alignment with a single-digit number. I know that your code will work for two digits as well. However, not testing for it means - this code needs to be manually tested every time it is changed. Manual testing is time-consuming and error-prone.
However, _testing all kind of alignment is hard_.
Everytime you are faced with a difficult test, ask yourself if there is off-the-shelf functionality that does the job better.  E.g., could we output comma-separated values (csv) instead? It's easy to test for the presence of commas. The alignment doesn't need a test - excel will do it for you :)